Add an icon with the song type

Matthieu-LAURENT39 opened the pull request:
This simply adds a small icon next to the song's name matching the song's type.  
This is useful to get the type at a glance, especially since right now the only way to see it is (as far as i'm aware) to go in the "change song type" modal.

I didn't add any icon for the "Original" song type, since that will be the one used on most songs anyways. 
That way icons are only shown for the few songs that have a different type, avoiding clutter.

Also note that the icons used right now are probably not the best (especially for acoustic and a capella), but i wanted to reuse Iconsax as much as possible. It would probably be a good idea to change a few of these for something from another icon set though.

## Song view
![Song view](https://github.com/user-attachments/assets/c3975a58-e4ee-4d27-874d-e6316c8c0f0c) 

## Album view
![Album view](https://github.com/user-attachments/assets/a37b3f22-4182-4780-b7c1-beb9ce437244) |

Matthieu-LAURENT39 commented:
I do agree, it does feel a bit visually loaded (especially in the song view), and even more so since that information can be deduced from the song's name.  
You're right that it would probably fit better in the track / song info, i'll try to move it later today.

For the icons, would you want to change some? I feel this is as good as it gets with the Iconsax set, but i'm not sure if it's worth using another set just for a few icons.

Arthi-chaud commented:
Ah thanks, that's very much happreciated.

Regarding the icons, I don't want to rely on another dependency just for that. Unless you have some SVGs that you think might fit better (and manage to use them with Nextjs), I think we should stick to Iconsax.
I would replace the icon for 'edit' with the `Cut` one, and use the `Edit` icon for demos.

Matthieu-LAURENT39 commented:
I moved the icons to the song info page and the track info modal. 
Since they aren't being displayed in a prominent location, i added an icon for the "Original" song type for consistency sake.
I also changed the icons for cut and edit, per your suggestion.

![image](https://github.com/user-attachments/assets/fd1b7973-7575-4a5c-9327-44b6197e3737)
![image](https://github.com/user-attachments/assets/99f4de2b-6f02-465e-bfdf-67d352ab694c)

Arthi-chaud commented:
Thanks! I'll review this ASAP

Arthi-chaud commented on front/src/components/icons.tsx:
```diff
@@ -91,3 +91,16 @@ export const MinusIcon = Iconsax.Minus;
+export const SongTypeInstrumentalIcon = Iconsax.Musicnote;
+export const SongTypeEditIcon = Iconsax.Cut;
+export const SongTypeCleanIcon = Iconsax.TickCircle;
+export const SongTypeDemoIcon = Iconsax.Edit;
```
Using this icon feels weird, it looks like an edit icon (as if we could change the song type).
Could you use `ReceiptEdit` instead?

Arthi-chaud commented on front/src/components/list-item/item.tsx:
```diff
@@ -24,6 +24,7 @@ import {
 	ListItemText,
 	ListItem as MUIListItem,
 	Skeleton,
+	Stack,
```
I think you can revert all the changes made to this file, as we agreed that icons shouldn't appear here

Arthi-chaud commented on front/src/components/list-item/song-item.tsx:
```diff
@@ -125,6 +125,7 @@ const SongItem = <
 				/>
 			}
 			title={song?.name}
+			titleIcon={undefined}
```
You can remove that too

Arthi-chaud commented on front/src/components/release-tracklist.tsx:
```diff
@@ -28,6 +28,7 @@ import {
 	ListItemText,
 	ListSubheader,
 	Skeleton,
+	Stack,
```
I think you can revert all the changes made to this file, as we agreed that icons shouldn't appear here

Arthi-chaud commented on front/src/components/song-grid.tsx:
```diff
@@ -55,6 +55,7 @@ const SongGrid = ({ songs, parentArtistName }: SongGridProps) => {
 							/>
 						}
 						title={song?.name}
+						titleIcon={undefined}
```
Same

Arthi-chaud commented on front/src/pages/playlists/[slugOrId]/index.tsx:
```diff
@@ -202,6 +202,7 @@ const PlaylistEntryItem = ({ entry, onClick }: PlaylistEntryItemProps) => (
 			/>
 		}
 		title={entry?.name}
+		titleIcon={undefined}
```
Same

Arthi-chaud commented on front/src/components/track-file-info.tsx:
```diff
@@ -39,11 +42,19 @@ const TrackFileInfo = ({ trackId }: { trackId: number }) => {
 		name: track.data?.name,
+		type: track.data?.type,
+		songType: songType && (
+			<Stack direction="row" spacing={1} alignItems="center">
```
These two type-cast are useless here

Arthi-chaud commented on front/src/components/song-type-icon.tsx:
```diff
@@ -0,0 +1,69 @@
+    SongTypeRemixIcon
+} from "./icons";
+
+const SongTypeIcon = ({ type, size = 20 }: { type: SongType; size?: number }) => {
```
I think we can remove that component: the tooltip is not useful in both places it's used in, and I don't like hard-coding paddings and positions. What you can do instead is something like:

```ts
const SongTypeIcon = ({ type, ...props }: {type: SongType } & IconProps) => {
    const Icon = () => { 
       switch (type) {
           case 'Original': return SongTypeOriginalIcon;
           // Do the other cases
       }
    }
    return <Icon {...props}/>
}
```

The paddings and alignments should be done at the places where you use the icons

Arthi-chaud commented on front/src/components/track-file-info.tsx:
```diff
@@ -39,11 +42,19 @@ const TrackFileInfo = ({ trackId }: { trackId: number }) => {
 	> = {
 		name: track.data?.name,
+		type: track.data?.type,
+		songType: songType && (
```
There is an edge case to consider: the track may not have a song (e.g. for a documentary video).
If `track.data.song === null`, it should display `N/A` (see the BPM for an example)

Arthi-chaud commented on front/src/pages/songs/[slugOrId]/[...tab].tsx:
```diff
@@ -179,6 +180,25 @@ const SongPage: Page<GetPropsTypesFrom<typeof prepareSSR>> = ({ props }) => {
 								song.data && `${song.data?.name} (${t("info")})`
 							}
 						/>
+						{song.data?.type && (
```
Could you do something similar to the BPM? I.e. When the data is loading, show a skeleton.
The `Song Type` label should always appear though. Something like:

```ts
// Pseudo code
<Stack row>
  <Typography>{songType}</Typopgraphy>
  {song.data ? <Stack row>{/*show icon and song type*/}</Stack> : <Skeleton />}
</Stack>
```

Arthi-chaud reviewed (changes requested):
- [x] `<SongTypeIcon/>` should not do stuff regarding the layout.
- [ ] You can remove all the changes that affect the song grid/list 
- [x] A few fixes with the skeleton and the track not always having a song type

Matthieu-LAURENT39 commented on front/src/i18n/translations/fr.json:
```diff
@@ -257,5 +257,6 @@
-	"emptyStateHome": "Oh wow, c'est silencieux par ici.."
-}
+	"emptyStateHome": "Oh wow, c'est silencieux par ici..",
+	"songType": "Type de Titre"
```
"Type du titre" does sound a lot more natural in this context.
I assume i should keep the current capitalization (i.e. "Type du Titre")? Since all the nearby text is also title-cased.

Matthieu-LAURENT39 commented:
I think that covers all the requested changes
Sorry for the messy code in this PR, i'm pretty new to frontend work '^^

Arthi-chaud commented:
@Matthieu-LAURENT39 Can you make sure the code type-checks please?

And could you run Biome after that? (with `yarn lint` and `yarn format`)

Matthieu-LAURENT39 commented:
@Arthi-chaud Should be good now!

Arthi-chaud commented:
LGTM! Thanks again for your contribution!
